### PR TITLE
fix Block#toObject API

### PR DESCRIPTION
**Pull request by maraoz**
This PR fixes https://github.com/bitpay/bitcore/issues/1180
and standardizes a little more the classes `Block` and `BlockHeader`

### CODE REVIEW

**Comment by braydonf** [comment]
That should fix the issue, one thing that we may want to do, and may need to be a part of another PR, is to use strings for the hashes and merkleRoot everywhere except in `toBuffer` and `fromBuffer`. That way the reversing only needs to be dealt with during serialization.

**Comment by pnagurny** [comment]
I agree with Braydon. This will fix the issue here and at some point you may want to consider just using strings everywhere except where serialized.

**Comment by maraoz** [comment]
I agree with your idea, but let's do it in another PR :)

File: lib/transaction/transaction.js [hunk 1]
```diff
@@ -313,6 +313,7 @@ Transaction.prototype.toObject = function toObject() {
     obj.changeIndex = this._changeIndex;
   }
   if (!_.isUndefined(this._fee)) {
+    console.log(this._fee);
```
Comment (braydonf) [inline lib/transaction/transaction.js]: leftover `console.log`

**Comment by braydonf** [comment]
Bump to 0.12 for this?

File: lib/transaction/transaction.js (hunk 1 above)
Comment (maraoz) [inline lib/transaction/transaction.js]: @braydonf the sniper. thanks

File: lib/block/block.js [hunk 2]
```diff
@@ -62,27 +51,35 @@ Block._from = function _from(arg) {
 };
 
 /**
- * @param {String|Object} - A JSON string or object
+ * @param {String} - A JSON string
  * @returns {Object} - An object representing block data
  * @private
  */
 Block._fromJSON = function _fromJSON(data) {
```
Comment (braydonf) [inline lib/block/block.js]: I think many fromJSON methods currently handle both an object and string, we should make sure these are all consistent.

Comment (maraoz) [inline lib/block/block.js]: Reading the code I realized many classes don't have a `fromObject` method. I removed the Object option on those because I added a `fromObject` method. IMO we should have separate `fromJSON`/`fromObject` methods in each class. `fromJSON` most commonly will be equivalent to `Foo.fromObject(JSON.parse(input))`

**Comment by maraoz** [comment]
Regarding bumping minor version: agreed. I'll release this Wednesday.

File: lib/block/block.js (hunk 2 above)
Comment (braydonf) [inline lib/block/block.js]: I'd say we should make this a part of another PR that we can go through add those methods where necessary, so that there there are round trips for `fromJSON/toJSON` and `fromObject/toObject`. However if either of those handle both a string and object, we only need one pair.
